test/src/sdk-harness/test_artifacts/reentrancy_target_contract/src/main.sw

```diff
@@ -23,7 +23,7 @@ impl Target for Contract {
         if is_reentrant() {
             true
         } else {
-            let result: Result<Sender, AuthError> = msg_sender();
+            let result: Result<Identity, AuthError> = msg_sender();
```

Comment:
Why is this change needed? Is the SDK harness not testing this?

Reply:
Forgot to mention: as a side effect this PR also exposes some errors that were being thrown out before. This should be one of those.